components/renku_data_services/namespace/db.py

```diff
@@ -262,12 +265,17 @@ async def update_group(
                 )
             session.add(schemas.NamespaceOldORM(slug=group.namespace.slug, latest_slug_id=group.namespace.id))
             group.namespace.slug = new_slug_str
+            slug_changed = True
         if patch.name is not None:
             group.name = patch.name
         if patch.description is not None:
             group.description = patch.description if patch.description else None
 
-        return group.dump()
+        new_group = group.dump()
+
+        if slug_changed:
+            return models.GroupUpdate(old=old_group, new=new_group)
```

Comment:
I think I would then always return `GroupUpdate`. It is a bit confusing that `update_group(…)` returns only an `GroupUpdate` when the slug changed, otherwise only the new group. Since we have the data now anyways, I think it is more consistent in general to always return the `GroupUpdate` no matter what changed. Wdyt?